File: Linked_Lists/Easy/21/21.py
from typing import Optional

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        head = ListNode()
        temp = head
        while list1 and list2:
            if list1.val < list2.val:
                head.next = ListNode(list1.val)
                head = head.next
                list1 = list1.next
            else:
                head.next = ListNode(list2.val)
                head = head.next
                list2 = list2.next
        if list1:
                head.next = list1
        else:
                head.next = list2
        return temp.next

File: Linked_Lists/Easy/21/21.c
// Definition for singly-linked list.
struct ListNode {
    int val;
    struct ListNode *next;
};

struct ListNode* mergeTwoLists(struct ListNode* list1, struct ListNode* list2){
    struct ListNode *head = (struct ListNode *)malloc(sizeof(struct ListNode));
    struct ListNode *ptr = head;
    while(list1 && list2){
        if(list1->val < list2->val){
            head->next = list1;
            list1 = list1->next;
        }else{
            head->next = list2;
            list2 = list2->next;
        }
        head = head->next;
    }
    if(list1){
        head->next = list1;
    }else{
        head->next = list2;
    }
    return ptr->next;
}